### PR TITLE
API integration for FaceID and IP Adapters

### PR DESCRIPTION
**Pull request by Trojaner**
## Description
Adds `ip_adapter` and `face_id` fields to txt2img and img2img API requests. 

The difference to calling these via `alwayson_scripts` (IP Adapter) and `script_name` (FaceID) is that they are clearly documented this way and doesn't need guessing what the parameters are and how they are aligned.

Screenshot from Swagger:
image

I also fixed the ipadapter and faceid scripts to support base64 encoded images instead of just PIL images and added debug logs for all HTTP exceptions.

## Notes
Basically transforms ip_adapter into alwayson_scripts and face_id into script_name and script_args unless there are already conflicting values set.

## Environment and Testing
Windows

Tested and confirmed working with diffusers backend. Isn't supported on original backend anyway.

### CODE REVIEW

**Comment by vladmandic** [comment]
i don't like hardcoded sanitize for specific arg order - instead of just
> request.alwayson_scripts["IP Adapter"]["args"][2] = "<base64 encoded image>"

can you make `prepare_img_gen_request` be a general function that sanitizes all such possible strings?
basically ignore base64, if type is str and len >1k, replace str in req with `<str len(field)>`

**Comment by Trojaner** [comment]
Will do on weekend (if I dont forget)
